### DISCUSSION TITLE
Anyway to get all station data at once?

### DISCUSSION THREAD

**Discussion by EricWardner** [weather-gov/api#657] [Q&A] [answered]
I'm trying to pull all the current temperature data and wondering if there's any place I can query all the station data at once or if I have to hit every single station endpoint. Thanks!

**Comment by sullynole** [comment, marked as answer]
If you want many different observations at once, it may be better to grab the observations data from the upstream source: 

https://madis-data.ncep.noaa.gov

For example, here is the text/xml viewer option, which accepts some query parameters:
https://madis-data.ncep.noaa.gov/public/sfcdumpguest.shtml

**Reply by EricWardner** [reply]
I think this is going to be perfect! Thanks!

**Reply by EricWardner** [reply]
@sullynole Do you know of way to query or get info about a station? Like lat/long at a minimum?

I found this page: https://[madis-data.ncep.noaa.gov/madisPublic1/data/stations/stntbl.txt](https://madis-data.ncep.noaa.gov/madisPublic1/data/stations/stntbl.txt) but it doesn't seem to have all the site (like METAR for example)

**Reply by sullynole** [reply]
It is broken down by each data provider here: https://madis-data.ncep.noaa.gov/madisPublic1/data/stations/

https://madis-data.ncep.noaa.gov/madisPublic1/data/stations/METARTable.txt
